teachertool/src/types/index.ts

```diff
@@ -50,9 +50,15 @@ export type RequestStatus = "init" | "loading" | "error" | "success";
 export type ProjectData = pxt.Cloud.JsonScript & {
     inputText: string;
 };
+
 export type ConfirmationModalOptions = {
     title: string;
     message: string;
     onCancel: () => void;
     onContinue: () => void;
 };
+
+export type CriteriaTemplateSegment = {
```

Comment:
For better or worse, I've been advocating for centralizing shared types here.

Reply:
Yeah, my internal logic here was to use `criteria.ts` for types that "compose" a criteria object, so `CatalogCriteria` which uses `CriteriaParameter` for a field inside it, which in turn uses `CriteriaParameterType` for a field inside it, and so on... The template segment exists outside of that, so I didn't put it in the file.

Admittedly, I was lazy and put the `CriteriaInstance` types inside `criteria.ts` as well, so it's all a bit muddled. I honestly don't feel that strongly, but the `index.ts` feels like a better "default" to me than `criteria.ts`.

Reply:
Yeah, that's fine, I don't feel strongly about this. I feel like if that's the pattern we want to follow, then maybe the `EvaluationStatus` enum and `CriteriaResult` interface should be moved out then, too. I put them there to start because I thought anything revolving around Criteria should be in `criteria.ts`, but it sounds like we don't want that to be the case. It isn't something to do for this PR, but something we can do in the future maybe.